flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java

```diff
@@ -258,7 +259,7 @@ public void run() {
                 // over
                 if (records == null) {
                     try {
-                        records = consumer.poll(pollTimeout);
+                        records = consumer.poll(Duration.ofMillis(pollTimeout));
```

Comment:
Note:  This changes the blocking behavior of this call.

https://cwiki.apache.org/confluence/display/KAFKA/KIP-266%3A+Fix+consumer+indefinite+blocking+behavior

I'd like to see a proper ticket for this change.

Reply:
yes, sure
jira issue created https://issues.apache.org/jira/browse/FLINK-30056

Reply:
please let me know if i need to change commit message

Reply:
done